### PR TITLE
actually call near-vm when using VMKind::NearVm

### PR DESCRIPTION
**Pull request by Ekleog-NEAR**

### CODE REVIEW

File: runtime/near-vm-runner/src/near_vm_runner.rs [hunk 1]
```diff
@@ -9,53 +9,55 @@ use near_primitives::contract::ContractCode;
...
     bulk_memory: WASM_FEATURES.bulk_memory,
     multi_value: WASM_FEATURES.multi_value,
     tail_call: WASM_FEATURES.tail_call,
-    module_linking: WASM_FEATURES.module_linking,
     multi_memory: WASM_FEATURES.multi_memory,
     memory64: WASM_FEATURES.memory64,
     exceptions: WASM_FEATURES.exceptions,
+    mutable_global: true,
+    saturating_float_to_int: true,
+    sign_extension: true,
```
Comment (nagisa) [inline runtime/near-vm-runner/src/near_vm_runner.rs:44]: We went over this before, but double-checking these ideally happens before we roll something out (you mentioned believing these should be true, I still haven't double checked it, though.)

File: runtime/near-vm-runner/src/near_vm_runner.rs [hunk 2]
```diff
@@ -229,41 +231,41 @@ impl Wasmer2Config {
...
 
-pub(crate) struct Wasmer2VM {
+pub(crate) struct NearVmVM {
     pub(crate) config: VMConfig,
     pub(crate) engine: UniversalEngine,
 }
 
-impl Wasmer2VM {
-    pub(crate) fn new_for_target(config: VMConfig, target: wasmer_compiler::Target) -> Self {
+impl NearVmVM {
```
Comment (nagisa) [inline runtime/near-vm-runner/src/near_vm_runner.rs:254]: I think we can name this `NearVM`.

File: runtime/near-vm-runner/src/near_vm_runner.rs [hunk 3]
```diff
@@ -533,59 +533,144 @@ impl Wasmer2VM {
     }
 }
 
-impl wasmer_vm::Tunables for &Wasmer2VM {
+impl near_vm_vm::Tunables for &NearVmVM {
```
Comment (nagisa) [inline runtime/near-vm-runner/src/near_vm_runner.rs:536]: Similarly for this module.

**Review by nagisa — APPROVED** [review]

File: runtime/near-vm-runner/src/near_vm_runner.rs (hunk 3 above)
Comment (Ekleog-NEAR) [reply runtime/near-vm-runner/src/near_vm_runner.rs:536]: So this is a separate crate, that was copy-pasted from wasmer_vm ; if we rename it it’ll collide with the main near_vm crate (that was copy-pasted from the wasmer crate).

So yes, but it’ll have to wait until we merge the near-vm crates all together, as the crate split doesn’t make much sense for us

File: runtime/near-vm-runner/src/near_vm_runner.rs (hunk 1 above)
Comment (Ekleog-NEAR) [reply runtime/near-vm-runner/src/near_vm_runner.rs:44]: This message seems to confirm that these were on-by-default in previous versions of wasmparser: https://github.com/bytecodealliance/wasm-tools/pull/482

**Pull request merged**